src/NServiceBus.Persistence.NonDurable/Outbox/NonDurableOutboxPersistence.cs

```diff
@@ -39,13 +39,13 @@ public OutboxCleaner(NonDurableOutboxStorage storage, TimeSpan timeToKeepDedupli
                 nonDurableOutboxStorage = storage;
             }
 
-            protected override Task OnStart(IMessageSession session)
+            protected override Task OnStart(IMessageSession session, CancellationToken cancellationToken = default)
             {
                 cleanupTimer = new Timer(PerformCleanup, null, intervalToCheckForDuplicates, intervalToCheckForDuplicates);
                 return Task.CompletedTask;
             }
 
-            protected override Task OnStop(IMessageSession session)
+            protected override Task OnStop(IMessageSession session, CancellationToken cancellationToken = default)
             {
                 using (var waitHandle = new ManualResetEvent(false))
```

Comment:
Not in scope but I wonder if we should remove all timers in favour of the "task that spins + cancellation token source" approach we use pretty much everywhere else?